modules/partition-replicator/src/integrationTest/java/org/apache/ignite/internal/partition/replicator/ItZoneDataReplicationTest.java

```diff
@@ -397,6 +395,11 @@ void testDataRebalance(boolean truncateRaftLog) throws Exception {
 
         setPrimaryReplica(newNode, zonePartitionId);
 
+        // Wait for the data to appear. At the moment of writing, we don't have any partition safe time to wait for and
+        // the primary replica has been assigned manually, so there's no guarantee that the data has been replicated.
+        // Not using "assertTrue" on purpose, the next line will produce a nicer error message.
+        waitForCondition(() -> kvView1.getAll(null, data1.keySet()).equals(data1), 10_000L);
```

Comment:
Please add a todo with a link to corresponding ticket https://issues.apache.org/jira/browse/IGNITE-22620

Reply:
Thank you, I couldn't find the relevant ticket myself